sap-s4hana-cloud/2.2/modules/ROOT/pages/sap-s4hana-cloud-connector-upgrade-migrate.adoc

```diff
@@ -0,0 +1,80 @@
+= Upgrade SAP S/4HANA OData Connector to 2.x - Mule 4
+:page-aliases: connectors::sap/sap-s4hana-cloud-connector-upgrade-migrate.adoc
+
+Upgrade Anypoint Connector for SAP S/4HANA OData (SAP S/4HANA OData Connector) to take advantage of the increased security offered by using OAuth 2.0 authentication to connect to SAP S/4HANA OData.
+
+== Supported Upgrade Paths
+
+This is a direct upgrade from version 1.2.1 version to 2.x.x.
+
+[%header,cols="50a,50a"]
+|===
+|From Version | To Version
+|1.2.1 |2.x.x
+|===
+
+== Changes in Version 2.0.0
+
+* OAuth 2.0 is supported. +
+The *OAuth 2.0 Application Connection Provider* option has been added to *Connection* types. For information about the configuration parameters for this connection type, see xref:sap-s4hana-cloud-connector-reference.adoc[SAP S/4HANA OData Connector Reference].
+* Mule runtime engine (Mule) 4.2.1 or later is required.
+* The default value for the *Service directory* field has been removed.
+* Because the base package has been changed, all exported classes now have different fully qualified names.
+* The group ID of the connector’s JAR has changed.
+
+== Changes in 2.2.1
+
+SAP S/4HANA OData Connector now provides support for the OData APIs exposed by SAP S/4HANA S/4HANA on-premise.
+
+== Changes in 2.2.0
+
+You can now configure the response, client, and connection timeout parameters.
+
+== Requirements and Limitations
+
+Be aware of the following requirements and limitations before you upgrade to the latest version.
+
+=== Requirements
+
+You must use Mule 4.2.1 or later.
+
+=== Limitations
+
+This upgrade breaks backward compatibility with 1.x.x versions of the connector.
+
+== Upgrade Prerequisites
+
+Before you upgrade, create a backup of your files, data, and configurations in case you need to restore to them.
+
+== Upgrade Steps
+
+Follow these steps to perform the upgrade:
+
+. In Anypoint Studio, create a Mule project.
+. In *Mule Palette*, click *Search in Exchange*.
```

Comment:
I've been saying "the *Mule Palette* view because I think it sounds better. This is a little thing we can discuss later on.

Reply:
I think that's fine. I think that's how it is in the template.